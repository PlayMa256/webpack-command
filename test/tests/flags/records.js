const { apply, build, crcDist, test, validate } = require('../../util');

test('--records-*', module, () => {
  for (const name of ['input-output', 'path']) {
    const fixture = `records/records-${name}`;
    const opts = { fixture };

    let config;

    it(`${name} should validate`, () => {
      expect(validate(opts)).toEqual(true);
    });

    it(`${name} should apply`, () => {
      config = apply(opts);

      expect(config).toMatchSnapshot();
    });

    it(`${name} should build`, () =>
      build(config).then((result) => {
        expect(result).toMatchSnapshot();
        expect(crcDist()).toMatchSnapshot();
      }));
  }
});
