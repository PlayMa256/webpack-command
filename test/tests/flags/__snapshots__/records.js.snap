// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --records-* > input-output should apply #0 1`] = `
Object {
  "devtool": "source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "recordsInputPath": "<PROJECT_ROOT>/dist/records-input.json",
  "recordsOutputPath": "<PROJECT_ROOT>/dist/records-output.json",
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --records-* > input-output should build #0 1`] = `
"Version: webpack 4.3.0
      Asset      Size  Chunks             Chunk Names
    main.js  2.92 KiB    main  [emitted]  main
main.js.map  2.66 KiB    main  [emitted]  main
Entrypoint main = main.js main.js.map
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --records-* > input-output should build #1 1`] = `"f349b18e"`;

exports[`Flags > --records-* > path should apply #0 1`] = `
Object {
  "devtool": "source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "recordsPath": "<PROJECT_ROOT>/dist/records.json",
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --records-* > path should build #0 1`] = `
"Version: webpack 4.3.0
      Asset      Size  Chunks             Chunk Names
    main.js  2.92 KiB    main  [emitted]  main
main.js.map  2.66 KiB    main  [emitted]  main
Entrypoint main = main.js main.js.map
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --records-* > path should build #1 1`] = `"f349b18e"`;
